flink-core/src/main/java/org/apache/flink/api/connector/sink/GlobalCommitter.java

```diff
@@ -32,8 +32,11 @@
  *
  * @param <CommT> The type of information needed to commit data staged by the sink
  * @param <GlobalCommT> The type of the aggregated committable
+ * @deprecated Please use {@code WithPostCommitTopology} with {@code
+ *     StandardSinkTopologies#addGlobalCommitter}.
```

Comment:
Since the `GlobalCommitter` became a class of operator in #18476, do we need a new interface design for the user to use the new e.g. `GlobalCommitOperator` at the interface level? Using `WithPostCommitTopology` described here means that the `GlobalCommitOperator` can only be used after a real `Committer`.

Reply:
I do not think we need a new interface for the global committer because so far it is only used rarely and mainly as a committer running with parallelism 1.
Theoretically, users can also use the global committer with `WithPreCommittingTopology`.

In the long run, I'd expect that more people start using the `WithPrecommitTopology` directly and less using the global committer. We just have to introduce an alternative to the existing `GlobalCommitter` interface therefore the helper method.